contracts/DestinationContractHandler.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier:MIT
+
+pragma solidity ^0.6.12;
+
+import "./Ownable.sol";
+
+abstract contract DestinationContractHandler is Ownable {
+    mapping(address => bool) public contracts;
+    address[] public acceptedContracts;
+
+    function acceptContract(address destinationContract) external onlyOwner {
+        require(
+            destinationContract != address(0),
+            "Contract cannot be zero address"
+        );
```

Comment:
The only one that can add a new address is the owner of the contract. We are assuming that prior to adding the new contract, he knows what he is adding.  But we can also include the validation.

Reply:
Having the validation makes it more robust, but if you have a validation on the API code that ultimately calls this function then it would be equivalent and adding the on-chain validation a bit redundant. If the owner doesn't use the API and calls the contract directly, then you can leave a comment in the contract about the expected prerequisites of the destinationContract address.

Reply:
A validation was included to make it more robust.